pkg/controllers/node/termination/controller.go

```diff
@@ -77,13 +79,24 @@ func (c *Controller) Finalize(ctx context.Context, node *v1.Node) (reconcile.Res
 	if !controllerutil.ContainsFinalizer(node, v1beta1.TerminationFinalizer) {
 		return reconcile.Result{}, nil
 	}
+
+	nodeGracePeriodExpirationTime, err := c.terminationGracePeriodExpirationTime(ctx, node)
```

Comment:
Does adding this taint come after `c.deleteAllNodeClaims` call if the terminationGracePeriodExpirationTime is based off of the NodeClaim's deletion timestamp?